authenc.c

```diff
@@ -314,7 +314,8 @@ static void read_tls_hash(struct scatterlist *dst_sg, int len, void *hash, int h
 
 static int pad_record(struct scatterlist *dst_sg, int len, int block_size)
 {
-	uint8_t pad[block_size];
+	uint8_t *pad;
+	pad = kmalloc_array(block_size,sizeof(*pad),GFP_KERNEL);
```

Comment:
1. Missing kfree().
2. Check coding style using checkpatch --strict.

It might be better avoiding dynamic memory allocation at runtime.
Considering TLS (padding + padding_length) is at most 256 bytes, I suggest keeping the variable on the stack and avoiding VLA by using pad[TLS_MAX_PADDING_SIZE] and define TLS_MAX_PADDING_SIZE = 256.

Reply:
@horiag Are you interested in fixing up this patch for the 1.11 release?